syntax = "proto3";
package pubsubs;

message ServerMessage{
    oneof Type{
        ClientDetails clientDetails = 1 ;
        Article publishArticle = 2 ;
        RequestMessage getArticle = 3 ;
    }
}

message ClientDetails {
	string client_uuid = 1 ;
    bool join = 2 ;
    bool leave = 3 ; 
}

message StatusOfClientRequest {
	bool request_status = 1;
}

message Article {
	oneof ArticleTags {
		bool SPORTS = 1;
		bool FASHION = 2;
		bool POLITICS = 3;
	}
	string author = 4;

	// time
	int32 day = 5;
	int32 month = 6;
	int32 year = 7;

	// actual content, word limit 200 characters
	string content = 8;

	// client details to verify clientelle
	string client_uuid = 9;
}

message RequestMessage {
	oneof Tags {
		bool allTags = 1;
		bool SPORTS = 2;
		bool FASHION = 3;
		bool POLITICS = 4;
	}
	string author = 5;

	// time
	int32 day = 6;
	int32 month = 7;
	int32 year = 8;

	// client details to verify clientelle
	string client_uuid = 9;
}

message ArticleList {
	repeated Article articles = 1;
	bool status = 2;
}